bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -548,6 +550,12 @@ private void configureCredentialProvider(StubSettings.Builder<?, ?> stubSettings
       stubSettings.setCredentialsProvider(
           FixedCredentialsProvider.create(
               buildCredentialFromPrivateKey(serviceAccount, keyFileLocation)));
+    } else if (!Strings.isNullOrEmpty(configuration.get(BIGTABLE_CUSTOM_CREDENTIALS_CLASS_KEY))) {
+      String customAuthClassName = configuration.get(BIGTABLE_CUSTOM_CREDENTIALS_CLASS_KEY);
```

Comment:
configuration.getClass(key, BigtableOAuthCredentials.class) and then update the method signatures appropriately

Reply:
Also why does this have the least precedence in the auth priority? I would think it belongs right after  BigtableExtendedConfiguration

Reply:
I don't think this is going to be a popular way to inject credentials. Customer will only use this method, when no other method will work. hence, its the last.